packages/sanity/src/core/studio/components/navbar/search/definitions/fields.ts

```diff
@@ -109,37 +117,38 @@ function getDocumentFieldDefinitions(
     prevFieldPath?: string
     prevTitlePath?: string[]
   }) {
-    const continueRecursion = depth <= MAX_OBJECT_TRAVERSAL_DEPTH
+    if (depth <= MAX_OBJECT_TRAVERSAL_DEPTH) {
```

Comment:
This should be:
```suggestion
    if (depth > MAX_OBJECT_TRAVERSAL_DEPTH) {
```
Since the old check was `!continueRecursion`